internal/integratedservices/services/ingress/ingressadapter/org_domain_service.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ingressadapter
+
+import (
+	"context"
+	"fmt"
+
+	"emperror.dev/errors"
+
+	"github.com/banzaicloud/pipeline/internal/integratedservices/services/ingress"
+	"github.com/banzaicloud/pipeline/src/auth"
+	"github.com/banzaicloud/pipeline/src/dns"
+)
+
+type OrgDomainService struct {
```

Comment:
I would stick to the best effort solution as seen in the posthook. This service will eventually go away/will be rewritten, so I don't think it's worth introducing the dependency. The best-effort solution is about 5 lines. If you want, you can wrap that in a service or something. It should probably be a cluster domain service though.

Reply:
How is this different from the implementation in the post-hook other than it's hidden behind a layer of abstraction to enabled easier testing?

Reply:
```
func (s OrgDomainService) GetOrgDomain(ctx context.Context, orgID uint) (ingress.OrgDomain, error) {
```

It binds the ingress service to a relatively old concept which will eventually be removed.

Reply:
The service's current implementation has to get these values somehow. When it won't, this all can be deleted. What's the harm? Would it be better to directly depend on an org repo service to get the organization's values and the base domain config value? Those concepts could become irrelevant just as easily.